packages/backend/src/core/UptimeUpdater.ts

```diff
@@ -0,0 +1,95 @@
+import { Logger, TaskQueue } from '@l2beat/common'
+import { ProjectUptime } from '@l2beat/config'
+import { EthereumAddress } from '@l2beat/types'
+
+import { ProjectInfo } from '../model'
+import { ApiMonitor } from '../peripherals/uptime/ApiMonitor'
+import { RpcMonitor } from '../peripherals/uptime/RpcMonitor'
+import { Clock } from './Clock'
+
+export class UptimeUpdater {
+  private taskQueue = new TaskQueue<void>(() => this.update(), this.logger)
+
+  constructor(
+    private rpcMonitor: RpcMonitor,
+    private apiMonitor: ApiMonitor,
+    private clock: Clock,
+    private projects: ProjectInfo[],
+    private logger: Logger,
+  ) {
+    this.logger = this.logger.for(this)
+  }
+
+  start() {
+    this.logger.info('Started')
+    return this.clock.onNewMinute(() => {
+      this.taskQueue.addToFront()
+    })
+  }
+
+  async update() {
+    this.logger.info('Update started')
+
+    await Promise.allSettled(
+      this.projects.flatMap(({ urls }) =>
+        urls?.map((url) => this.checkUptime(url)),
+      ),
+    )
+
+    this.logger.info('Update completed')
+  }
+
+  async checkUptime(uptime: ProjectUptime) {
+    if (uptime.action === 'rpc_estimateGas') {
```

Comment:
I am not a fan of this list of ifs. Ideally somehow the Updater would receive a mapping of action type to handlers and just pick from the mapping